app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala

```diff
@@ -25,37 +25,39 @@ import wco.datamodel.wco.declaration_ds.dms._2._
 
 class BorderTransportMeansBuilder @Inject()(countriesService: CountriesService) extends ModifyingBuilder[ExportsDeclaration, Declaration] {
   override def buildThenAdd(model: ExportsDeclaration, t: Declaration): Unit = {
-    val transportMeans = new Declaration.BorderTransportMeans()
     val transport = model.transport
-    val hasDeparture = transport.hasDepartureTransportCode
+
+    val hasTransportLeavingTheBorder = transport.hasTransportLeavingTheBorder
     val hasBorder = transport.hasBorderTransportDetails
-    if (hasDeparture || hasBorder) {
-      if (hasDeparture) {
-        appendDepartureTransport(transport, transportMeans)
-      }
-      if (hasBorder) {
-        appendBorderTransport(transport, transportMeans)
-      }
+    val hasDepartureTransport = transport.hasDepartureTransportDetails
+
+    if (hasTransportLeavingTheBorder || hasBorder || hasDepartureTransport) {
+      val transportMeans = new Declaration.BorderTransportMeans
+
+      if (hasTransportLeavingTheBorder) appendTransportLeavingTheBorder(transport, transportMeans)
+
+      if (hasBorder) appendBorderTransport(transport, transportMeans)
+      else if (hasDepartureTransport) appendDepartureTransport(transport, transportMeans)
```

Comment:
This bit seems a little confusing as both `appendBorderTransport` and `appendDepartureTransport` actually sets both BorderTransport and DepartureTransport values when called (but this `if` makes it seems like they are mutually exclusive).

Reply:
They both set BorderTransport only, but `appendDepartureTransport` is called **only** if the user did not enter any value in BorderTransport (and in this case `appendBorderTransport` is not called). Note that is Builder is dedicated to BorderTransport only.

Reply:
Ok, I see. Maybe `appendDepartureTransport` could be named `appendBorderTransportWithDepartureValue' to make this more obvious?